controllers/neutronapi_controller.go

```diff
@@ -1032,8 +1032,12 @@ func (r *NeutronAPIReconciler) reconcileNormal(ctx context.Context, instance *ne
 		return ctrl.Result{}, err
 	}
 
-	if instance.Status.ReadyCount > 0 {
-		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	if instance.Status.ReadyCount > 0 &&
+		depl.GetDeployment().Generation <= depl.GetDeployment().Status.ObservedGeneration {
```

Comment:
@mrkisaolamb as per [1] I think this is the right check to do, right?

[1] https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go#L712

Reply:
@fmount 
Yes, this seems to be the most common approach for guarding against unexpected resets and re-creations